src/lib/include/simd/simd_doubles.h

```diff
@@ -0,0 +1,162 @@
+#ifndef DISTOPIA_SIMD_DOUBLE_H
+#define DISTOPIA_SIMD_DOUBLE_H
+
+#include <cassert>
+#include <immintrin.h>
+
+// class to wrap __m128d 2 x packed doubles
+class SimdDoubleX2 {
+public:
+  inline void set(const double d);
+  inline void load(const double *source);
+  inline void store(double *target);
+  inline void loadU(const double *source);
+  inline void storeU(double *target);
+  inline void zero();
+  inline void reciprocal();
+
+  __m128d contents;
+};
+
+// set to all same value
+inline void SimdDoubleX2::set(const double d) { contents = _mm_set1_pd(d); }
+
+// aligned load from double*
+inline void SimdDoubleX2::load(const double *source) {
+  // check alignment
+  assert(size_t(source) % 16 == 0);
+  contents = _mm_load_pd(source);
+}
+
+// aligned store to double*
+inline void SimdDoubleX2::store(double *target) {
+  // check alignment
+  assert(size_t(target) % 16 == 0);
+  _mm_store_pd(target, contents);
+}
+
+// unaligned load from double*
+inline void SimdDoubleX2::loadU(const double *source) {
+  contents = _mm_loadu_pd(source);
+}
+
+// unaligned store from double*
+inline void SimdDoubleX2::storeU(double *target) {
+  _mm_storeu_pd(target, contents);
+}
+
+// zero
+inline void SimdDoubleX2::zero() { contents = _mm_setzero_pd(); }
+
+// take reciprocal (1/contents)
+// max relative error ~ 0.000366
+inline void SimdDoubleX2::reciprocal() {
+  contents = _mm_cvtps_pd(_mm_rcp_ps(_mm_cvtpd_ps(contents)));
```

Comment:
This is waaaay too inaccurate. It negates the entire point of using doubles.

Reply:
I really don’t see a need for a fast reciprocal function. We use division sparingly anyway, so I’m pretty sure it’s a red herring.

Reply:
ill pull out